In [1]:
import numpy as np
X = np.genfromtxt("ionosphere.txt", delimiter=",")
X[:3]

array([[ 1.     ,  0.     ,  0.99539, -0.05889,  0.85243,  0.02306,
         0.83398, -0.37708,  1.     ,  0.0376 ,  0.85243, -0.17755,
         0.59755, -0.44945,  0.60536, -0.38223,  0.84356, -0.38542,
         0.58212, -0.32192,  0.56971, -0.29674,  0.36946, -0.47357,
         0.56811, -0.51171,  0.41078, -0.46168,  0.21266, -0.3409 ,
         0.42267, -0.54487,  0.18641, -0.453  ,  1.     ],
       [ 1.     ,  0.     ,  1.     , -0.18829,  0.93035, -0.36156,
        -0.10868, -0.93597,  1.     , -0.04549,  0.50874, -0.67743,
         0.34432, -0.69707, -0.51685, -0.97515,  0.05499, -0.62237,
         0.33109, -1.     , -0.13151, -0.453  , -0.18056, -0.35734,
        -0.20332, -0.26569, -0.20468, -0.18401, -0.1904 , -0.11593,
        -0.16626, -0.06288, -0.13738, -0.02447, -1.     ],
       [ 1.     ,  0.     ,  1.     , -0.03365,  1.     ,  0.00485,
         1.     , -0.12062,  0.88965,  0.01198,  0.73082,  0.05346,
         0.85443,  0.00827,  0.54591,  0.00299,  0.83775, -0.13644

In [2]:
X = np.genfromtxt("ionosphere.txt", delimiter=",",
    usecols=np.arange(34))
X[:3]

array([[ 1.     ,  0.     ,  0.99539, -0.05889,  0.85243,  0.02306,
         0.83398, -0.37708,  1.     ,  0.0376 ,  0.85243, -0.17755,
         0.59755, -0.44945,  0.60536, -0.38223,  0.84356, -0.38542,
         0.58212, -0.32192,  0.56971, -0.29674,  0.36946, -0.47357,
         0.56811, -0.51171,  0.41078, -0.46168,  0.21266, -0.3409 ,
         0.42267, -0.54487,  0.18641, -0.453  ],
       [ 1.     ,  0.     ,  1.     , -0.18829,  0.93035, -0.36156,
        -0.10868, -0.93597,  1.     , -0.04549,  0.50874, -0.67743,
         0.34432, -0.69707, -0.51685, -0.97515,  0.05499, -0.62237,
         0.33109, -1.     , -0.13151, -0.453  , -0.18056, -0.35734,
        -0.20332, -0.26569, -0.20468, -0.18401, -0.1904 , -0.11593,
        -0.16626, -0.06288, -0.13738, -0.02447],
       [ 1.     ,  0.     ,  1.     , -0.03365,  1.     ,  0.00485,
         1.     , -0.12062,  0.88965,  0.01198,  0.73082,  0.05346,
         0.85443,  0.00827,  0.54591,  0.00299,  0.83775, -0.13644,
         0.75535, 

In [3]:
y = np.genfromtxt("ionosphere.txt", delimiter=",",
usecols=34, dtype='int')
y[:3]

array([ 1, -1,  1])

In [4]:
def cube(x):
    """this is my first function in this module; it cubes a number"""
    return x**3
cube(3)


27

In [5]:
help(cube)

Help on function cube in module __main__:

cube(x)
    this is my first function in this module; it cubes a number



In [6]:
def powers(x):
    """this function computes both square and cube"""
    return x**2, x**3
powers(3)


(9, 27)

In [7]:
powers(3)[1]

27

In [8]:
a = np.array([1,2,3])
a

array([1, 2, 3])

In [9]:
b = a
b

array([1, 2, 3])

In [10]:
a[1] = 100
a

array([  1, 100,   3])

In [11]:
b

array([  1, 100,   3])

In [12]:
b = np.array(a)

In [13]:
b = a.copy()

In [14]:
a = np.array([1,2,3])
b = np.array([5,6])
np.concatenate((a,b))

array([1, 2, 3, 5, 6])

In [15]:
my_array = np.concatenate((np.arange(10**6),np.array([4])))

___
Exercise 1:

In [16]:
my_array

array([     0,      1,      2, ..., 999998, 999999,      4])

my_array is an array containing value from 0 to 999999 with step of 1, and a integer 4 at the end of the array.
___

Exercise 2:

In [17]:
help(np.concatenate)

Help on _ArrayFunctionDispatcher in module numpy:

concatenate(...)
    concatenate((a1, a2, ...), axis=0, out=None, dtype=None, casting="same_kind")

    Join a sequence of arrays along an existing axis.

    Parameters
    ----------
    a1, a2, ... : sequence of array_like
        The arrays must have the same shape, except in the dimension
        corresponding to `axis` (the first, by default).
    axis : int, optional
        The axis along which the arrays will be joined.  If axis is None,
        arrays are flattened before use.  Default is 0.
    out : ndarray, optional
        If provided, the destination to place the result. The shape must be
        correct, matching that of what concatenate would have returned if no
        out argument were specified.
    dtype : str or dtype
        If provided, the destination array will have this dtype. Cannot be
        provided together with `out`.

        .. versionadded:: 1.20.0

    casting : {'no', 'equiv', 'safe', 'same_kind', 

The repeat parentheses {concatenate(k=(0,...,69))} is there to define the set k as a tuple of elements, in order to apply the method successfully

___

In [18]:
import time
start = time.time()
print(sum(4 >= my_array))
print(time.time() - start,"seconds")

6
0.19640636444091797 seconds


In [19]:
start = time.time()
count = 0
for n in range(my_array.size):
    if 4 >= my_array[n]:
        count =  count + 1
print(count)
print(time.time() - start,"seconds")

6
0.4516315460205078 seconds


___
Exercise 3:

In [20]:
array = []
for n in range(my_array.size):
    if 4 >= my_array[n]:
        array.append((my_array[n],n))
print(array)

[(0, 0), (1, 1), (2, 2), (3, 3), (4, 4), (4, 1000000)]


Only the first five element generated from np.arrange and the last element from np.array([4]) has a value =< 4

___
Exercise 4:

In [21]:
import math
start = time.time()
current_minimum = [math.inf,0]
current_maximum = [-math.inf,0]
for n in range(my_array.size):
    if current_minimum[0] > my_array[n]:
        current_minimum[0] = my_array[n]
        current_minimum[1] = n
    if current_maximum[0] < my_array[n]:
        current_maximum[0] = my_array[n]
        current_maximum[1] = n
print("minimum= (%d, %d)" % (current_minimum[0],current_minimum[1]))
print("maximum= (%d, %d)" % (current_maximum[0],current_maximum[1]))
print(time.time() - start,"seconds")

minimum= (0, 0)
maximum= (999999, 999999)
1.0645158290863037 seconds


___
Exercise 5:

In [22]:
def squared_norm(array,length):
    value = 0
    for n in range(length):
        value += array[n] * array[n]
    return value
squared_norm([3,4,5,6,7],5)

135

___
Exercise 6:

In [23]:
def norm_squared(array):
    value = 0
    for n in range(len(array)):
        value += array[n] * array[n]
    return value
norm_squared([3,4,5,6,7])

135

___
Exercise 7:

In [24]:
def norm(array):
    return math.sqrt(norm_squared(array))
k=norm([3,4,5,6,7])
p=k**2
print("%f^2 = %f" % (k,p))

11.618950^2 = 135.000000


___